### PR TITLE
Polynomial features will support degree from 0 to n

### LINKED ISSUE #273: fixup scikit-learn PolynomialFeatures
In #269, the basics are implemented, but to be fully complete we need:
(1) degree larger than 2
(2) support for  interaction_only
(3) more tests

### PR DESCRIPTION
**Pull request by giriprasad51**
#273 fixed

import hummingbird
import numpy as np
from sklearn.preprocessing import PolynomialFeatures
from hummingbird.ml import convert
for degree in range(10):
  X = np.arange(15).reshape(3,5)
  y = np.random.randint(2, size=6)

  poly = PolynomialFeatures(degree)
  poly_sk = poly.fit_transform(X)
  poly_convert = convert(poly, 'pytorch')
  poly_h = poly_convert.transform(X)
  if (poly_sk != poly_h).all() :
    print(degree)

### CODE REVIEW

File: tests/test_sklearn_poly_features_converter.py [hunk 1]
```diff
@@ -48,11 +48,11 @@ def test_sklearn_poly_featurizer_raises(self):
 
         # TODO: delete when implemented
         model = PolynomialFeatures(degree=4, include_bias=True, order="F").fit(data)
-        self.assertRaises(NotImplementedError, hummingbird.ml.convert, model, "torch")
+        self._test_sklearn_polynomial_featurizer(data, model)
 
         # TODO: delete when implemented
```
Comment (ksaur) [inline tests/test_sklearn_poly_features_converter.py]: Can you please remove the TODOs on lines 49 and 53?

File: hummingbird/ml/operator_converters/sklearn/poly_features.py [hunk 2]
```diff
@@ -74,8 +72,8 @@ def convert_sklearn_poly_features(operator, device, extra_config):
     if operator.raw_operator.interaction_only:
         raise NotImplementedError("Hummingbird does not currently support interaction_only flag for PolynomialFeatures")
 
-    if operator.raw_operator.degree != 2:
-        raise NotImplementedError("Hummingbird currently only supports degree 2 for PolynomialFeatures")
+    if operator.raw_operator.degree < 0:
+        raise NotImplementedError("Hummingbird not supports negtive degree for PolynomialFeatures")
```
Comment (ksaur) [inline hummingbird/ml/operator_converters/sklearn/poly_features.py]: Can you please change to "Hummingbird does not support negative degree for PolynomialFeatures"

File: hummingbird/ml/operator_converters/sklearn/poly_features.py [hunk 3]
```diff
@@ -19,40 +20,37 @@ class PolynomialFeatures(PhysicalOperator, torch.nn.Module):
...
-        else:
-            return torch.cat([x_orig, x], dim=1)
+            bias = torch.ones(x.size()[0], 1, device=device)
+            features.append(bias)
+
+        # Generate polynomial features
+        for d in range(1, self.degree + 1):
+            for combo in itertools.combinations_with_replacement(range(self.n_features), d):
+                if self.interaction_only and len(set(combo)) != d:
+                    continue
```
Comment (ksaur) [inline hummingbird/ml/operator_converters/sklearn/poly_features.py:49]: These 2 lines (48-49) are missing from the [code coverage](https://app.codecov.io/gh/microsoft/hummingbird/pull/763/blob/hummingbird/ml/operator_converters/sklearn/poly_features.py#L49).   Can you think of an example that would cover it?

**Review by ksaur — CHANGES_REQUESTED** [review]
Thank you so much for this contribution!! Just a few minor things. :)

**Comment by giriprasad51** [comment]
Hii @ksaur! As per your suggestion I made some changes in the code please go through it. Thanks!

**Review by ksaur — APPROVED** [review]
Thank you for your contributions! :)

**Pull request merged**